arlo-client/src/components/Audit/Setup/BallotManifest/index.tsx

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState, useEffect } from 'react'
+import { Formik, FormikProps, Form } from 'formik'
+import { useParams } from 'react-router-dom'
+import { toast } from 'react-toastify'
+import styled from 'styled-components'
+import { HTMLSelect, FileInput, H4 } from '@blueprintjs/core'
+import { IErrorResponse } from '../../../../types'
+import FormWrapper from '../../../Atoms/Form/FormWrapper'
+import FormButtonBar from '../../../Atoms/Form/FormButtonBar'
+import FormButton from '../../../Atoms/Form/FormButton'
+import schema from './schema'
+import { ErrorLabel } from '../../../Atoms/Form/_helpers'
+import FormSection, {
+  FormSectionDescription,
+} from '../../../Atoms/Form/FormSection'
+import { api, checkAndToast } from '../../../utilities'
+import { useAuthDataContext } from '../../../UserContext'
+import { IJurisdictionsFileResponse } from '../../useSetupMenuItems/getJurisdictionFileStatus'
+
+export const Select = styled(HTMLSelect)`
+  margin-top: 5px;
+`
+
+interface IValues {
+  csv: File | null
+}
+
+const BallotManifest: React.FC<{}> = () => {
+  const { electionId } = useParams()
+  const { meta } = useAuthDataContext()
+  const { jurisdictions } = meta!
+  const [file, setFile] = useState<IJurisdictionsFileResponse>({
+    file: null,
+    processing: null,
+  })
+  const [showFileUpload, setShowFileUpload] = useState(true)
+  useEffect(() => {
+    try {
+      ;(async () => {
+        const fileResponse:
+          | IJurisdictionsFileResponse
+          | IErrorResponse = await api(
+          `/election/${electionId}/jurisdiction/${jurisdictions[0].id}/ballot-manifest`
+        )
+        // checkAndToast left here for consistency and reference but not tested since it's vestigial
+        /* istanbul ignore next */
+        if (checkAndToast(fileResponse)) return
+        setFile(fileResponse)
+        if (fileResponse.file) setShowFileUpload(false)
+      })()
+    } catch (err) /* istanbul ignore next */ {
+      // TEST TODO
+      toast.error(err.message)
+    }
+  }, [electionId, jurisdictions, setFile, setShowFileUpload])
+  const submit = async (values: IValues) => {
+    try {
+      /* istanbul ignore else */
+      if (values.csv) {
+        const formData: FormData = new FormData()
+        formData.append('manifest', values.csv, values.csv.name)
+        const errorResponse: IErrorResponse = await api(
+          `/election/${electionId}/jurisdiction/${jurisdictions[0].id}/ballot-manifest`, // do we need a selector here?
+          {
+            method: 'PUT',
+            body: formData,
+          }
+        )
+        if (checkAndToast(errorResponse)) return
+      }
+    } catch (err) /* istanbul ignore next */ {
```

Comment:
one thing to note is that because the file processing happens in the background, this `PUT` won't return any errors with the processing. those will be available from the `GET` endpoint under `processing` once the processing completes. wondering if we want to poll for those and show them (or, easier, we could just show them on page load and not do any polling, then the user would have to refresh to see errors)

Reply:
:thinking: I think we could (in another PR) implement a polling functionality where it gives a loading indicator while it is processing, then shows a green check mark when it has completed processing, and a red x with errors if something wrong happened.